vendor/riscv/riscv-isa-sim/customext/cv32a6embedded.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2023 Thales DIS France SAS
+//
+// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.0.
+//
+// Original Author: Zbigniew CHAMSKI <zbigniew.chamski@thalesgroup.com>
+
+#define DECODE_MACRO_USAGE_LOGGED 1
+#include "csrs.h"
+#include "processor.h"
+#include "cva6.h"
+#include <cstdlib>
+
+// This class instantiates the CV32A6 "embedded" control&status extension.
+class cv32a6_embedded_extn_t : public cva6_extn_t
+{
+ public:
+  const char* name() { return "cv32a6embedded"; }
+
+  cv32a6_embedded_extn_t()
+  {
+  }
+  ~cv32a6_embedded_extn_t() {}
+
+  void reset()
+  {
+    std::cerr << "[Extension '" << name() << "'] reset()" << std::endl;
+
+    // Complain if processor is not set.
+    if (!p || !p->get_state()) {
+      std::cerr << "ERROR: processor/state not set in cv32a6_embedded_extn_t::reset()!" << std::endl;
+      exit(1);
+    }
+
+    state_t *state = p->get_state();
+    
+    // Create ICACHE CSR.  Only bit 0 is writable. Initialize to 1.
+    reg_t icache_mask = ICACHE_EN;
+    state->csrmap[CSR_ICACHE] = icache_reg = std::make_shared<masked_csr_t>(p, CSR_ICACHE, icache_mask, 1);
```

Comment:
Could we add both ICACHE and DCACHE here?

Reply:
OK for me, but we need to freeze a feature set for this PR (#2334) so that can be marked "complete" and merged.  From there on, the Spike extension can be enriched with new features following the basic pattern.